aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cu

```diff
@@ -152,4 +193,100 @@ SparseTensor coalesce_sparse_cuda(const SparseTensor& self) {
   return dst;
 }
 
+Tensor sparse_mask_helper_cuda(
+    const SparseTensor& t,
+    const Tensor& mask_indices) {
+  /*
+    This is a helper function which filter values from `t._values()` using the
+    `mask_indices`. This CUDA implementation uses `thrust::lower_bound`
+    operation to find the intersection of the `mask_indices` and the
+    `t._indices()` to then filter the values.
+
+    Inputs:
+      `t`             - coalesced sparse tensor input
+      `mask_indices`  - mask indices tensor
+  */
+  TORCH_CHECK(t.is_sparse(), "t: input is not a sparse tensor");
+  TORCH_CHECK(t.is_coalesced(), "t:  input is uncoalesced");
+  TORCH_CHECK(mask_indices.dim() == t._indices().dim(), "mask_indices: operands have incompatible indices dim; self has dim ",
+      t._indices().dim(), " but mask has dim ", mask_indices.dim());
+  TORCH_CHECK(mask_indices.is_contiguous(), "mask_indices: mask is not contiguous");
+
+  int64_t r_nnz = mask_indices.size(1);
+  auto t_values = t._values().contiguous();
+  auto full_size = t.sizes();
+  auto vsize = t_values.sizes().vec();
+  vsize[0] = r_nnz;
+
+ 
+  if (t.sparse_dim() == 0) {
+    Tensor t_values_expand = t_values;
+    t_values_expand = t_values_expand.expand(vsize).clone(at::MemoryFormat::Contiguous);
```

Comment:
you don't need `.clone()` here, just `.contiguous()` will do?

Reply:
Yes, just contiguous